pkg/watermark/processor/processor_manager.go

```diff
@@ -263,11 +274,20 @@ func (v *ProcessorManager) startTimeLineWatcher() {
 					continue
 				}
 				if otValue.Idle {
-					p.offsetTimeline.PutIdle(otValue)
+					if v.opts.isReduce {
+						p.offsetTimelines[0].PutIdle(otValue)
+					} else {
+						p.offsetTimelines[otValue.Partition].PutIdle(otValue)
+					}
 					v.log.Debugw("TimelineWatcher- Updates", zap.String("bucket", v.otWatcher.GetKVName()), zap.Int64("idleWatermark", otValue.Watermark), zap.Int64("idleOffset", otValue.Offset))
 				} else {
 					// NOTE: currently, for source edges, the otValue.Idle is always false
-					p.offsetTimeline.Put(otValue)
+					// for reduce we will always have only one partition
+					if v.opts.isReduce {
```

Comment:
what about source?

Reply:
source always publishes the wmb with partition set to 0